integration-provisioning/src/main/java/org/symphonyoss/integration/provisioning/service/KeyPairService.java

```diff
@@ -175,6 +181,21 @@ private void generateCertificate(Application application, String appKeyFilename,
     }
 
     executeProcess(genPKCS12Command);
+
+    setFileOwnership(appCertFilename, FILE_OWNERSHIP_USER, FILE_OWNERSHIP_GROUP);
+    setFileOwnership(appPKCS12Filename, FILE_OWNERSHIP_USER, FILE_OWNERSHIP_GROUP);
+  }
+
+  /**
+   * Sets the ownership of the given file with according to the user
+   */
+  private void setFileOwnership(String filename, String user, String group) {
+    LOGGER.info("Setting ownership for {} ({}:{})", filename, user, group);
+
+    String changeOwnerCommand =
+        String.format(CHANGE_OWNER_CMD, user, group, filename);
+
+    executeProcess(changeOwnerCommand);
```

Comment:
You might use NIO Java API to perform this operation rather than execute a SO system call.

Docs:
http://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#setOwner(java.nio.file.Path,%20java.nio.file.attribute.UserPrincipal)
http://docs.oracle.com/javase/7/docs/api/java/nio/file/attribute/PosixFileAttributeView.html
http://stackoverflow.com/questions/13241967/change-file-owner-group-under-linux-with-java-nio-files